Source/DafnyCore/DafnyPrelude.bpl

```diff
@@ -188,26 +190,26 @@ axiom (forall bx : Box ::
 // generated programmatically. Except, Bv0 is given here.
 axiom (forall bx : Box ::
     { $IsBox(bx, TBitvector(0)) }
-    ( $IsBox(bx, TBitvector(0)) ==> $Box($Unbox(bx) : Bv0) == bx && $Is($Unbox(bx) : Set Box, TBitvector(0))));
+    ( $IsBox(bx, TBitvector(0)) ==> $Box($Unbox(bx) : Bv0) == bx && $Is($Unbox(bx) : Bv0, TBitvector(0))));
```

Comment:
This rewriting is surprising, I would have expected this instead:
```suggestion
    ( $IsBox(bx, TBitvector(0)) ==> $Box($Unbox(bx) : Bv0) == bx && $Is($Unbox(bx) : Set, TBitvector(0))));
```

Reply:
Huh. That is weird. I'll give it a closer look.

Reply:
I thought `Set Box` looked like a copy-paste error, but maybe I am missing something...

Reply:
You're probably right, but this mean that this particular type cast was probably untested then until now.

Reply:
Yes, maybe it makes more sense to have that as another PR with proper tests, not to have this as a hidden change here.

Reply:
Not having a test for now is ok since it was not tested anyway, but if you had a test, indeed, you could move it to another PR. For now, let's piggy back on this PR to perform this bug fix if it's correct.